metrics-collectd/src/test/java/com/codahale/metrics/collectd/CollectdReporterTest.java

```diff
@@ -0,0 +1,159 @@
+package com.codahale.metrics.collectd;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import org.collectd.api.ValueList;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import static com.codahale.metrics.collectd.ValueListAssert.assertThat;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class CollectdReporterTest {
+
+    @ClassRule
+    public static Receiver receiver = new Receiver(25826);
+
+    private MetricRegistry registry;
+    private CollectdReporter reporter;
+
+    @Before
+    public void setUp() {
+        registry = new MetricRegistry();
+        reporter = CollectdReporter.forRegistry(registry).build(new Collectd("localhost", 25826));
+    }
+
+    @Test
+    public void handlesCounter() throws Exception {
+        Counter counter = registry.counter("api.rest.requests");
+        counter.inc(42);
+
+        reporter.report();
+
+        ValueList values = receiver.next();
+        assertNotNull(values.getHost());
+        assertEquals("api", values.getPlugin());
+        assertEquals("rest", values.getPluginInstance());
+        assertEquals("requests", values.getType());
+        assertThat(values.getTypeInstance()).isEmpty();
+        assertEquals(42, values.getValues().get(0).intValue());
+    }
+
+    @Test
+    public void handlesMeter() throws Exception {
+        Meter meter = registry.meter("api.rest.requests");
+        meter.mark(112);
+
+        reporter.report();
+
+        ValueList values = receiver.next();
+        assertNotNull(values.getHost());
+        assertEquals("api", values.getPlugin());
+        assertEquals("rest", values.getPluginInstance());
+        assertEquals("requests", values.getType());
+        assertEquals("count", values.getTypeInstance());
+        assertThat(receiver.next()).hasType("requests", "m1_rate");
+        assertThat(receiver.next()).hasType("requests", "m5_rate");
+        assertThat(receiver.next()).hasType("requests", "m15_rate");
+        assertThat(receiver.next()).hasType("requests", "mean_rate");
+    }
+
+    @Test
+    public void handlesGauge() throws Exception {
+        Gauge<Double> gauge = registry.register("gauge", new Gauge<Double>() {
```

Comment:
We can create a `Gauge` with a lambda expression as `(Gauge<Double>) () -> 0.25`. Should be more concise and readable.